tmt/steps/provision/docker.py

```diff
@@ -0,0 +1,140 @@
+    """
+    Docker provision plugin
+    """
+
+    # Guest instance
+    _guest = None
+
+    # Supported keys
+    _keys = ["image", "container", "pull"]
+
+    # Supported methods
+    _methods = [tmt.steps.Method(name="docker", doc=__doc__, order=50)]
+
+    @classmethod
+    def options(cls, how=None):
+        """ Prepare command line options for connect """
+        return [
+            click.option(
+                "-i",
+                "--image",
+                metavar="IMAGE",
+                help="Select image to use. Short name or complete url.",
+            ),
+            click.option(
+                "-c",
+                "--container",
+                metavar="NAME",
+                help="Name or id of an existing container to be used.",
+            ),
+            click.option(
+                "-p",
+                "--pull",
+                is_flag=True,
+                help="Force pulling a fresh container image.",
+            ),
+        ] + super().options(how)
+
+    def wake(self, options=None, data=None):
+        """ Override options and wake up the guest """
+        if data:
+            self._guest = GuestDocker(data, name=self.name, parent=self.step)
+            self._guest.run_container(self.get("image"))
+
+    def go(self):
+        """ Provision the container """
+        super().go()
+        # Prepare data for the guest instance
+        data = dict()
+        for key in self._keys:
+            data[key] = self.get(key)
+        self._guest = GuestDocker(data, name=self.name, parent=self.step)
+        self._guest.run_container(
+            self.get("image"),
+            path=str(Path("/data/tests/integration/inhibit-if-kmods-is-not-supported")),
+        )
+
+    def guest(self):
+        """ Return the provisioned guest """
+        return self._guest
+
+    def requires(self):
+        """ List of required packages needed for workdir sync """
+        return GuestDocker.requires()
+
+    def show(self):
+        """ Show provision details """
+        super().show(self._keys)
+
+
+class GuestDocker(tmt.Guest):
+    """ Docker """
+
+    def ansible(self, playbook):
+        """ Prepare docker container with ansible playbook """
+        playbook = self._ansible_playbook_path(playbook)
+        stdout, stderr = self.run(
+            f"ansible-playbook {self._ansible_verbosity()} "
+            f'--extra-vars "container={self.container}" '
+            f"{playbook} "
+            f"-vvv"
+        )
+        self._ansible_summary(stdout)
+
+    def run_container(self, image, path):
+        container, _ = self.run(
+            f"docker run -itd --workdir={path} {image}",
+            message=f"Running container from the image {image}",
+        )
+        self.container = container.strip()
+
+    def execute(self, command, **kwargs):
+        """ Execute command on docker """
+        # Change to given directory on guest if cwd provided
```

Comment:
no longer valid as the container runs with --workdir pointing to the test path. However this might be wrong if there are tests in different locations (not the case of convert2rhel)